beancount/parser/macros.h

```diff
@@ -0,0 +1,33 @@
+#ifndef BEANCOUNT_PARSER_MACROS_H
+#define BEANCOUNT_PARSER_MACROS_H
+
+#define _CC_FUNC_01(FN, X, ...) FN(X)
+#define _CC_FUNC_02(FN, X, ...) FN(X); _CC_FUNC_01(FN, __VA_ARGS__)
+#define _CC_FUNC_03(FN, X, ...) FN(X); _CC_FUNC_02(FN, __VA_ARGS__)
+#define _CC_FUNC_04(FN, X, ...) FN(X); _CC_FUNC_03(FN, __VA_ARGS__)
+#define _CC_FUNC_05(FN, X, ...) FN(X); _CC_FUNC_04(FN, __VA_ARGS__)
+#define _CC_FUNC_06(FN, X, ...) FN(X); _CC_FUNC_05(FN, __VA_ARGS__)
+#define _CC_FUNC_07(FN, X, ...) FN(X); _CC_FUNC_06(FN, __VA_ARGS__)
+#define _CC_FUNC_08(FN, X, ...) FN(X); _CC_FUNC_07(FN, __VA_ARGS__)
+#define _CC_FUNC_09(FN, X, ...) FN(X); _CC_FUNC_08(FN, __VA_ARGS__)
+#define _CC_FUNC_10(FN, X, ...) FN(X); _CC_FUNC_09(FN, __VA_ARGS__)
+
+#define _CC_FUNC_SEQ(_01, _02, _03, _04, _05, _06, _07, _08, _09, _10, NAME, ...) NAME
```

Comment:
How does this work? I keep staring at it, don't grok it.
You call it below, passing __VA_ARGS__ as the first argument, I'm assuming this gets spliced into the call to __CC_FUNC_SEQ(), and then ... well that just eats up the arguments and expands to NAME... which is...what? 
This is simultaneously magic and beautiful yet incomprehensible.
Can you add a comment explaining how this is expanded?

Reply:
Indeed it takes a while to see how this works, but once you see it, it is obvious. It would be easier to find an article that describes the technique, but I haven't found one in a rapid search. I'll either add some comments in the form of an explanation or a link to one.

Reply:
Gimme some comment on this and we're good to go. Ideally a link...

Reply:
I didn't find a link to an explanation I liked. Thus I wrote one. Let me know if it is understandable.